element-connector/src/main/java/org/eclipse/californium/elements/RawData.java

```diff
@@ -330,4 +330,16 @@ public boolean isSecure() {
 		return (correlationContext != null &&
 				correlationContext.get(DtlsCorrelationContext.KEY_SESSION_ID) != null);
 	}
+	
+	/**
+	 * Callback, when context gets available. Used on sending an message.
+	 * 
+	 * @param context established context to be forwarded to the callback.
+	 */
+	public void onContextEstablished(CorrelationContext context) {
```

Comment:
we need either this method or `getMessageCallback()`, I guess.
Can you check whether `getMessageCallback()` is used for any other purpose than notifying it about the context?
If that is the case I would rather go with `onContextEstablished()` ...